remote/rpc/compat.go

```diff
@@ -71,6 +72,25 @@ func contains(ss []string, s string) bool {
 	return false
 }
 
+// listServicesRolloutStatus polyfills the rollout status.
+func listServicesRolloutStatus(ss []v6.ControllerStatus) {
+	for i := range ss {
+		// Polyfill for daemons that list pod information in status ('X out of N updated')
+		if n, _ := fmt.Sscanf(ss[i].Status, "%d out of %d updated", &ss[i].Rollout.Updated, &ss[i].Rollout.Desired); n == 2 {
+			// Daemons on an earlier version determined the workload to be ready if updated == desired.
+			//
+			// Technically, 'updated' does *not* yet mean these pods are ready and accepting traffic. There
+			// can still be outdated pods that serve requests. The new way of determining the end of a rollout
+			// is to make sure the desired count equals to the number of 'available' pods and zero outdated
+			// pods. To make older daemons reach a "rollout is finished" state we set 'available', 'ready',
+			// and 'updated' all to the same count.
```

Comment:
Great explanation, thank you for including this :scroll: 

Does this mean older daemons will give an inaccurate measure of rollout progress, since they conflate updated and available? I am OK with this -- just checking I've understood.

Reply:
that's exactly what it means